### PR TITLE
Move ParseConfirmTarget from rpc/mining to rpc/util

### PR DESCRIPTION
**Pull request by ryanofsky**
Util is a better home since it's called both by wallet and mining code.

Suggested https://github.com/bitcoin/bitcoin/pull/15288#discussion_r254449444

### CODE REVIEW

**Comment by promag** [comment]
utACK c706207.

File: src/rpc/util.cpp [hunk 1]
```diff
@@ -4,9 +4,11 @@
 
 #include <key_io.h>
 #include <keystore.h>
+#include <policy/fees.h>
 #include <rpc/protocol.h>
 #include <rpc/util.h>
 #include <tinyformat.h>
+#include <validation.h>
```
Comment (practicalswift) [inline src/rpc/util.cpp]: Nit: Sort includes :-)

Comment (ryanofsky) [reply src/rpc/util.cpp]: Thanks, fixed now

**Review by ryanofsky — COMMENTED** [review]
Updated c7062070c15e5fec47b740164a8a5b9d8508883b -> 50e647210d513dab8f40ded61fe439ecaa9a67c3 ([pr/movtarget.1](https://github.com/ryanofsky/bitcoin/commits/pr/movtarget.1) -> [pr/movtarget.2](https://github.com/ryanofsky/bitcoin/commits/pr/movtarget.2), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/movtarget.1..pr/movtarget.2)) to fix #include sort order

**Comment by laanwj** [comment]
Agree—utACK 50e647210d513dab8f40ded61fe439ecaa9a67c3

**Comment by jnewbery** [comment]
utACK 50e647210d513dab8f40ded61fe439ecaa9a67c3. Thanks Russ.

The `#include <rpc/mining.h>` can be removed from `rpcwallet.cpp` once `generate()` has been removed post v0.18 (was deprecated in #14468)

**Pull request merged**